### PR TITLE
Implement starknet_core_types::curve module

### PR DESCRIPTION
**Pull request by pefontana**
# Implement starknet_core_types::curve module
* Move the Felt implementation to starknet_core_types
* Implement Implement starknet_core_types::curve module:
  * affine_point
  * projective_point 
  * curve_errors

## Does this introduce a breaking change?
Yes

### CODE REVIEW

**Review by diegokingston — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,5 @@ Cargo.lock
 # Added by cargo
 
 /target
+
+.DS_Store
```
Comment (MauroToscano) [inline .gitignore:21]: Doesn't need to be pushed

**Review by MauroToscano — APPROVED** [review]

File: crates/starknet-types-core/src/curve/projective_point.rs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    /// The point at infinity.
+    pub fn identity() -> ProjectivePoint {
+        Self(ShortWeierstrassProjectivePoint::neutral_element())
+    }
+
+    /// Creates the same point in affine coordinates. That is,
+    /// returns [x / z: y / z: 1] where `self` is [x: y: z].
+    /// Panics if `self` is the point at infinity.
```
Comment (entropidelic) [inline crates/starknet-types-core/src/curve/projective_point.rs:26]: small detail: The doc says it panics but a `Result` is returned

**Review by entropidelic — APPROVED** [review]

File: crates/starknet-types-core/README.md [hunk 3]
```diff
@@ -1,16 +1,33 @@
...
 
-`starknet-types-core` is a crate focusing on Starknet types related to computation and execution. This crate is part of an initiative to standardize the representation of the `Felt` type in Rust, reducing code complexity and improving performance across the Starknet Rust ecosystem.
+Core types representation for Starknet.
 
-The types in this crate require performance and optimization for specific arithmetic and cryptographic operations, making it ideal for computational tasks within the Starknet ecosystem.
+## Overview
+
+The `starknet-types-core` crate provides:
+* The universal `Felt` (Field Element) type for Cairo and STARK proofs. It was created to reduce the fragmentation in the Starknet Rust ecosystem by providing a standardized representation of the `Felt` type.
+* The `AffinePoint` and `ProjectivePoint` structs, that represent a point in the Stark curve with which you can operate
```
Comment (fmoletta) [inline crates/starknet-types-core/README.md:9]:
```suggestion
* The `AffinePoint` and `ProjectivePoint` structs, that represent a point in the Stark curve with which you can operate.
```
Maybe perform elliptic curve operations sound better?

File: crates/starknet-types-core/README.md [hunk 4]
```diff
@@ -1,16 +1,33 @@
...
+- Standardized `Felt` type: Simplify your codebase by using our standardized `Felt` type.
+- Optimized for performance: The `Felt` type has been optimized for high-performance applications.
+
+## Examples
+
+Here are some examples of how to use the `Felt` type:
+
+```rust
+    let x = Felt::from(18)
+```
```
Comment (fmoletta) [inline crates/starknet-types-core/README.md:22]: Should't there be at least a couple more simple examples?

File: .gitignore (hunk 1 above)
Comment (pefontana) [reply .gitignore:21]: Sure, 
restored!

File: crates/starknet-types-core/src/curve/projective_point.rs (hunk 2 above)
Comment (pefontana) [reply crates/starknet-types-core/src/curve/projective_point.rs]: Nice Catch!

File: crates/starknet-types-core/README.md (hunk 3 above)
Comment (pefontana) [reply crates/starknet-types-core/README.md]: Yes, sounds better!
Done

File: crates/starknet-types-core/README.md (hunk 4 above)
Comment (pefontana) [reply crates/starknet-types-core/README.md]: Done

**Comment by pefontana** [comment]
Merged 
https://github.com/starknet-io/types-rs/pull/8

**Pull request closed without merging**